File: backend/src/main/java/com/jnsmarter/iptvbackend/model/Channel.java
package com.jnsmarter.iptvbackend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "channels")
public class Channel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "playlist_id", nullable = false)
    private Playlist playlist;

    @Column(nullable = false)
    private String name;

    private String logoUrl;
    private String groupTitle;
    private String tvgId;

    @Column(length = 2000, nullable = false)
    private String streamUrl;

    private int channelNumber;

    @Column(nullable = false)
    private boolean favorite = false;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Playlist getPlaylist() { return playlist; }
    public void setPlaylist(Playlist playlist) { this.playlist = playlist; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getLogoUrl() { return logoUrl; }
    public void setLogoUrl(String logoUrl) { this.logoUrl = logoUrl; }

    public String getGroupTitle() { return groupTitle; }
    public void setGroupTitle(String groupTitle) { this.groupTitle = groupTitle; }

    public String getTvgId() { return tvgId; }
    public void setTvgId(String tvgId) { this.tvgId = tvgId; }

    public String getStreamUrl() { return streamUrl; }
    public void setStreamUrl(String streamUrl) { this.streamUrl = streamUrl; }

    public int getChannelNumber() { return channelNumber; }
    public void setChannelNumber(int channelNumber) { this.channelNumber = channelNumber; }

    public boolean isFavorite() { return favorite; }
    public void setFavorite(boolean favorite) { this.favorite = favorite; }
}

File: jn-smarter-iptv-player-pro-flutter/jn-smarter-iptv-player-pro/frontend_flutter/lib/models/channel.dart
class Channel {
  final int id;
  final String name;
  final String logoUrl;
  final String groupTitle;
  final String tvgId;
  final String streamUrl;
  final int channelNumber;
  final bool favorite;
  final int playlistId;

  Channel({
    required this.id,
    required this.name,
    required this.logoUrl,
    required this.groupTitle,
    required this.tvgId,
    required this.streamUrl,
    required this.channelNumber,
    required this.favorite,
    required this.playlistId,
  });

  factory Channel.fromJson(Map<String, dynamic> json) => Channel(
        id: json['id'] as int,
        name: json['name'] as String? ?? 'Unnamed channel',
        logoUrl: json['logoUrl'] as String? ?? '',
        groupTitle: json['groupTitle'] as String? ?? 'Uncategorized',
        tvgId: json['tvgId'] as String? ?? '',
        streamUrl: json['streamUrl'] as String,
        channelNumber: json['channelNumber'] as int? ?? 0,
        favorite: json['favorite'] as bool? ?? false,
        playlistId: json['playlistId'] as int,
      );
}
